reader.go

```diff
@@ -1737,11 +1752,26 @@ func (r *reader) run(ctx context.Context, offset int64) {
 
 				switch {
 				case offset < first:
+					// We've probably tried to read an offset that has been deleted,
+					// e.g. because the message exceeded the topic's retention policy.
+					// ReaderConfig.AutoOffsetReset controls where the reader seeks to.
+					seekTo := first
+					if r.autoOffsetReset == LastOffset {
+						seekTo = last
```

Comment:
You can skip the reading of the `first` and `last` offset by using `-1` or `-2`.  There should be const for these already.

Reply:
Can you?  `first` and `last` are always read before this line executes.  Also having these values allows for the log message to indicate the correct number of messages skipped (`seekTo-offset`).

Reply:
The switch statement needs `first` and `last` for the cases. 

I do have a version of this code that calls `conn.Seek(r.autoOffsetReset, ...)`, but the behavior isn't much different than this code.

Reply:
So I don't think you should even call the `readOffset` and just jump right to the configured value. 
Let me see if I can find out what the java client does.

Reply:
https://github.com/apache/kafka/blob/0.10.2.1/clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java#L828

It looks like it just jumps when seeing that error.